docs/flow/modin/pandas/dataframe.rst

```diff
@@ -0,0 +1,110 @@
+:orphan:
+
+DataFrame Module Overview
+"""""""""""""""""""""""""
+
+Modin's ``pandas.DataFrame`` API
+''''''''''''''''''''''''''''''''
+Modin's ``pandas.DataFrame`` API is backed by a distributed object providing an identical
+API to pandas. After the user calls some ``DataFrame`` function, this call is internally
+rewritten into a representation that can be processed in parallel by the partitions. These
+results can be e.g., reduced to single output, identical to the single threaded
+pandas ``DataFrame`` method output.
+
+..
+    TODO: add link to the docs with detailed description of queries compilation
+    and execution ater DOCS-#2996 is merged.
+
+Public API
+----------
+
+.. autoclass:: modin.pandas.dataframe.DataFrame
+
+Usage Guide
+'''''''''''
+The most efficient way to create Modin ``DataFrame`` is to import data from external
+storage using the highly efficient Modin IO methods (for example using ``pd.read_csv``,
+see details for Modin IO methods in the :doc:`separate section </flow/modin/engines/base/io>`),
+but even if the data does not originate from a file, any pandas supported data type or
+``pandas.DataFrame`` can be used. Internally, the ``DataFrame`` data is distributed across all
```

Comment:
```suggestion
``pandas.DataFrame`` can be used. Internally, the ``DataFrame`` data is divided into
```

Reply:
Actually, `distributed across` seems to me more appropriate here than `divided into`, so let's ask another opinion. @devin-petersohn which option do you prefer?

Reply:
A partition implies a portion of the whole data so `distributed across` seems to me confused. Data can be distributed across nodes, but not across partitions themselves, I think.

Reply:
`distributed across partitions` doesn't make sense, but `distributed across CPUs and/or machines` or `divided into partitions` does. I prefer to keep the `distributed` language personally, but I think @amyskov can make the final decision.

Reply:
Ok, i see, so let's apply @YarShev suggestion.